astronomer/providers/core/operators/external_trigger_dagrun.py

```diff
@@ -0,0 +1,417 @@
+from __future__ import annotations
+
+import inspect
+import json
+import logging
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Sequence, cast
+
+import requests
+from airflow.exceptions import AirflowException, DagRunAlreadyExists
+from airflow.models import XCom
+from airflow.models.baseoperator import BaseOperator, BaseOperatorLink
+from airflow.models.dagrun import DagRun
+from airflow.providers.http.hooks.http import HttpHook
+from airflow.utils import timezone
+from airflow.utils.state import State
+from requests.auth import AuthBase, HTTPBasicAuth
+
+from astronomer.providers.core.triggers.external_dagrun import ExternalDeploymentDagRunTrigger
+from astronomer.providers.utils.typing_compat import Context
+
+XCOM_EXECUTION_DATE_ISO = "trigger_execution_date_iso"
+XCOM_DAG_ID = "trigger_dag_id"
+XCOM_RUN_ID = "trigger_run_id"
+
+if TYPE_CHECKING:
+    from airflow.models.taskinstance import TaskInstanceKey
+
+
+class ExternalDeploymentTriggerDagRunLink(BaseOperatorLink):
+    """
+    Operator link for ExternalDeploymentTriggerDagRunOperator.
+
+    It allows users to access DAG triggered by task using ExternalDeploymentTriggerDagRunOperator.
+    """
+
+    name = "Triggered DAG"
+
+    def get_link(self, operator: BaseOperator, *, ti_key: TaskInstanceKey) -> str:
+        """Provide link to browse externally triggered Dag run."""
+        # execution_date = XCom.get_value(ti_key=ti_key, key=XCOM_EXECUTION_DATE_ISO)
+        run_id = XCom.get_value(ti_key=ti_key, key=XCOM_RUN_ID)
+
+        operator = cast(ExternalDeploymentTriggerDagRunOperator, operator)
+        dag_id = operator.trigger_dag_id
+        http_conn_id = operator.http_conn_id
+        endpoint = DAGRUN_ENDPOINT.format(dag_id=dag_id, run_id=run_id)
+        hook = HttpHook(method="GET", http_conn_id=http_conn_id)
+
+        return hook.url_from_endpoint(endpoint=endpoint)
+
+
+class ExternalDeploymentTriggerDagRunOperator(BaseOperator):
+    """
+        External Deployment Trigger Dag Run Operator Make HTTP call to trigger a Dag Run and poll for the response state of externally
+        deployed DAG run to complete. The host should be external deployment url, header must contain access token
```

Comment:
```suggestion
        External Deployment Trigger Dag Run Operator makes an HTTP call to trigger a run of an externally deployed Dag and poll for its status asynchronously. The host should be the external deployment URL and the header must contain a valid access token
```

Reply:
Might need to check the indentation afterwards if you consider the suggestion :)